### PR TITLE
Migrate to new functional color system

### PR DESCRIPTION
**Pull request by colebemis**
## Background

An recent release of [Primer Primitives](https://primer.style/primitives) [deprecated many existing color variables](https://primer.style/primitives/colors#deprecated-variables) and introduced new [functional color variables](https://primer.style/primitives/colors#functional-variables).

## Problem

Primer React components is using deprecated functional color variables.

## Solution

1. Update `@primer/primitives` version so we can use the new functional color variables
2. Install `eslint-plugin-primer-react` to lint for usages of deprecated color variables
3. Autofix all usages of deprecated variables

---
Part of https://github.com/github/design-infrastructure/issues/1388

### CODE REVIEW

File: docs/content/theming.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 ---
 title: Theming
+description: Theming in Primer React is made possible by a theme object that defines your application's colors, spacing, fonts, and more.
 ---
 
-Theming in Primer React is made possible by a theme object that defines your application's colors, spacing, fonts, and more.
+
```
Comment (colebemis) [inline docs/content/theming.md:7]:
```suggestion
```

**Review by jonrohan — APPROVED** [review]
Changes look good 👍🏻

**Review by jfuchs — APPROVED** [review]
Overall looks good. Any idea why your migration didn't catch `src/Dialog.tsx#L88`?

**Comment by colebemis** [comment]
@jfuchs We don't have a suitable replacement `fade.black50` so the ESLint plugin didn't know what to replace it with

**Comment by colebemis** [comment]
@jonrohan What do you think we should replace `fade.black50` with here? https://github.com/primer/react/pull/1412/files/30e5c3a8eca7dabb2820c92bc1d16db24aad855b#diff-4c970416d228dbd4a03d585a97a9bd1f13cfe72cee9a25379d66c367cc812f68L88

**Pull request merged**